### PR TITLE
feat: add pageUrl macro

### PR DESCRIPTION
**Pull request by mister-ben**
Adds a `player.pageUrl` macro. This is intended as the URL at which the player has been embedded, whether directly or via an iframe. The name of the macro hopefully makes it clear that this is not an embed URL for the player itself.

If the player is in an iframe, this resolves to `document.referrer`, otherwise to `window.location.href`.

A test is only included for the iframe case for now, as the tests themselves run in an iframe.

### CODE REVIEW

**Comment by mister-ben** [comment]
Failing tests are unrelated

**Comment by misteroneill** [comment]
Failing tests are related to Firefox 66. I believe @alex-barstow is fixing that in another PR>

File: src/macros.js [hunk 1]
```diff
@@ -65,6 +65,9 @@ export default function adMacroReplacement(string, uriEncode, customMacros) {
   macros['{mediainfo.name}'] = this.mediainfo ? this.mediainfo.name : '';
   macros['{mediainfo.duration}'] = this.mediainfo ? this.mediainfo.duration : '';
   macros['{player.duration}'] = this.duration();
+  macros['{player.pageUrl}'] = (window.location !== window.parent.location) ?
```
Comment (misteroneill) [inline src/macros.js]: Maybe a comment here explaining that this conditional is used to detect iframes? It may be unfamiliar to some.

Comment (gkatsev) [reply src/macros.js]: I think that secure iframes throw if you access parent location, we'd want to double check and wrap in a try/catch if so

Comment (gkatsev) [reply src/macros.js]: Videojs has a isInIframe helper on the Dom until object

Comment (bcbclifford) [reply src/macros.js]: on chrome/ff/safari, `window.parent.location` doesn't throw, although `window.parent.location.href` does.

Comment (mister-ben) [reply src/macros.js]: didn't realise we had isInFrame. will use that instead then.

**Pull request merged**